design/ceph/node-loss-rbd-cephfs.md

```diff
@@ -0,0 +1,150 @@
+
+example of taint to be applied to lost node:
+
+```console
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoExecute
+# or
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoSchedule
+```
+
+> Note: This will be enabled by default in Rook if the NetworkFence CR is found, in the case for some reason user wants to disable this feature in Rook can edit `Rook-ceph-operator-config` configmap and update the `Rook_WATCH_FOR_NODE_FAILURE: "false"`.
+
+## How to get which IP to blocklist
+
+There are multiple networking options available for example, Host Networking, Pod networking, Multus etc. This make it difficult to know which NodeIP address to blocklist.
+For this we'll follow the following approach which will work for all networking options(but still will not work for external cluster which NodeIPs):
+
+1. Get the `volumesInUse` from the node which has the taint `out-of-service`.
+
+2. List all the pv and compare the pv `spec.volumeHandle` with the node `volumesInUse` field `volumeHandle`
+   1. Example:
+
+      Below is sample Node volumeInUse field
+
+    ```
+      volumesInUse:
+        - kubernetes.io/csi/rook-ceph.rbd.csi.ceph.com^0001-0009-Rook-ceph-0000000000000002-24862838-240d-4215-9183-abfc0e9e4002
+
+      # Note: The volumeInUse naming convention are `kubernetes.io/csi/ + CSI driver name + ^ + volumeHandle`
+    ```
+
+     and the following is pv `volumeInHandle`
+
+     ```
+      volumeHandle: 0001-0009-rook-ceph-0000000000000002-24862838-240d-4215-9183-abfc0e9e4002
+    ```
+
+3. For Ceph volumes on that node:
+
+   1. If RBD PVC makes use of the rbd status API
+      1. example:
+
+    ```console
+    $ rbd status <poolname>/<image_name>
+    Watchers:
+    watcher=172.21.12.201:0/4225036114 client.17881 cookie=18446462598732840961
+    ```
+
+   2. If CephFS PVC uses below CLI to clients connect to subvolume
+      1. example:
+
+      ```console
+      $ ceph tell mds.0 client ls
+      ...
+      ...
+      ...
+       "addr": {
+                "type": "v1",
+                "addr": "192.168.39.214:0",
+                "nonce": 1301050887
+       }
+      ...
+      ...
+      ...
+       ```
+
+4. Get IPs from step 3 (in above example `172.21.12.201`)
+5. blocklist the IP where the volumes are mounted.
+
+Suggested change
+Example of a NetworkFence CR that the Rook operator would create when a `node.kubernetes.io/out-of-service` taint is added on the node:
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: NetworkFence
+metadata:
+  name: <name> # We will keep the name the same as the node name
+  namespace: <ceph-cluster-namespace>
+spec:
+  driver: <driver-name> # for us it will `Rook-ceph.rbd.csi.ceph.com` or `Rook-ceph.cephfs.csi.ceph.com`
```

Comment:
```suggestion
  driver: <driver-name> #  extract the driver name from the PV object.
```
>for us it will `Rook-ceph.rbd.csi.ceph.com` or `Rook-ceph.cephfs.csi.ceph.com`

This will change based on the namespace where rook operator is running.